### PR TITLE
clean cluster role binding methods

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: clean cluster role binding methods

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

```release-note
NONE
```

### CODE REVIEW

File: api/pkg/handler/test/helper.go [hunk 1]
```diff
@@ -69,7 +69,7 @@ const (
 	UserID = "1233"
 	// UserID2 holds a test user ID
 	UserID2 = "1523"
-	// UserName holds a test user name
+	// UserEmail holds a test user name
```
Comment (kdomanski) [inline api/pkg/handler/test/helper.go]: ??

File: api/pkg/handler/v1/cluster/binding_test.go [hunk 2]
```diff
@@ -613,206 +597,3 @@ func TestListClusterRoleBinding(t *testing.T) {
 		})
 	}
 }
-
```
Comment (kdomanski) [inline api/pkg/handler/v1/cluster/binding_test.go:616]: Why are you deleting all these test?

File: api/pkg/handler/test/helper.go (hunk 1 above)
Comment (zreigz) [reply api/pkg/handler/test/helper.go]: my IDE refactor to many items automatically

File: api/pkg/handler/v1/cluster/binding_test.go (hunk 2 above)
Comment (kdomanski) [reply api/pkg/handler/v1/cluster/binding_test.go:616]: Ok, I see now that the relevant endpoints are gone.

Comment (zreigz) [reply api/pkg/handler/v1/cluster/binding_test.go:616]: we don't need those endpoints anymore. They were designed for a different view

Comment (zreigz) [reply api/pkg/handler/v1/cluster/binding_test.go:616]: I forgot to delete them from binding.go I will commit new changes.

**Comment by zreigz** [comment]
@kgroschoff @kdomanski PTAL

**Comment by kdomanski** [comment]
Since the updated `BindUserToClusterRoleEndpoint` now uses `client.Patch`, is it now possible to use this endpoint to update an existing role binding? If yes, then `TestPatchClusterRoleBinding` should probably remain (although updated).

File: api/pkg/handler/v1/cluster/binding.go [hunk 3]
```diff
@@ -920,12 +723,6 @@ func convertInternalRoleBindingsToExternal(roleBindings []rbacv1.RoleBinding) []
 
 func convertInternalClusterRoleBindingToExternal(clusterRoleBinding *rbacv1.ClusterRoleBinding) *apiv1.ClusterRoleBinding {
 	binding := &apiv1.ClusterRoleBinding{
-		ObjectMeta: apiv1.ObjectMeta{
```
Comment (kdomanski) [inline api/pkg/handler/v1/cluster/binding.go:923]: Why is this no longer needed?

**Comment by zreigz** [comment]
> Since the updated `BindUserToClusterRoleEndpoint` now uses `client.Patch`, is it now possible to use this endpoint to update an existing role binding? If yes, then `TestPatchClusterRoleBinding` should probably remain (although updated).

According to the new view we just need to bind/unbind user to some role. I will create a new method to unbind the user. In this case, we don't need `patch` endpoint and tests for it. It would be better to
add more tests for `TestBindUserToClusterRole` to cover scenarios:
- add already existing user
- add the user for existing binding to check is the subject list was expanded.
WDYT?

File: api/pkg/handler/v1/cluster/binding.go (hunk 3 above)
Comment (zreigz) [reply api/pkg/handler/v1/cluster/binding.go:923]: right now the ClusterRoleBinding is created automatically if doesn't exist during binding user to the role. It means Every Role contains exactly one binding which holds users with the role. To find this ClusterRoleBinding I need an only label and role name. The API model doesn't need to hold ClusterRoleBinding  ID and name in this case. We don't display binding name anywhere.

**Comment by zreigz** [comment]
@kdomanski I've added new test scenarios

**Pull request merged**